app/services/cocina_generator/structural/file_generator.rb

```diff
@@ -4,16 +4,20 @@ module CocinaGenerator
   module Structural
     # This generates a Cocina File for a work
     class FileGenerator
-      def self.generate(work_version:, attached_file:)
-        new(work_version:, attached_file:).generate
+      def self.generate(work_version:, attached_file:, resource_id:, cocina_file: nil)
+        new(work_version:, attached_file:, resource_id:,  cocina_file:).generate
       end
 
-      def initialize(work_version:, attached_file:)
+      ID_NAMESPACE = 'https://cocina.sul.stanford.edu'
+
+      def initialize(work_version:, attached_file:, resource_id:, cocina_file: nil)
```

Comment:
It looks like `resource_id` is a required argument even when the `cocina_file` is non-`nil`, but in that case it's never used. I'm wondering about ways to refactor this class to reflect that it has two modes---one that generates from a blob and another that generates from a Cocina instance---but I've not yet come up with anything that's better the current (small) number of branches where we need to discriminate between the two cases.

Reply:
I've made some tweaks to the `resource_id` and requiring both, as you suggested. 

I see what you're saying about there being two situations here to handle. Things already feel like they're getting pretty abstract, so I'm not sure about diving into that at this moment. But open to it, especially if this gets any more complicated.